spinalcordtoolbox/labels.py

```diff
@@ -278,24 +311,13 @@ def compute_mean_squared_error(img: Image, ref: Image) -> float:
     """
     coordinates_input = img.getNonZeroCoordinates()
     coordinates_ref = ref.getNonZeroCoordinates()
-
-    # check if all the labels in both the images match
-    if len(coordinates_input) != len(coordinates_ref):
-        raise ValueError(f"Input and reference image don't have the same number of labels! {len(coordinates_input)} vs {len(coordinates_ref)}")
-
-    rounded_coord_ref_values = [np.round(c.value) for c in coordinates_ref]
-    rounded_coord_in_values = [np.round(c.value) for c in coordinates_input]
     result = 0.0
 
+    # This line will add warning in the log if there are missing label.
+    find_missing_label(img, ref)
```

Comment:
The operation itself is pretty simple, so to fix the buggy behavior, it might be safer to just use something from [this StackOverflow question](https://stackoverflow.com/questions/35713093/how-can-i-compare-two-lists-in-python-and-return-not-matches) rather than making an entirely new function.

Reply:
You are right, I can modify find_missing_label, but I do think this is useful to have a function that returns the missing label. 
For example, to evaluate the sct_label_vertebrae, it would be easier to use this function than to go through each log file to check if there is a warning or not, as we would use `sct_run_batch` and perform prediction on 200 subjects (twice).

Reply:
> it would be easier to use this function than to go through each log file to check if there is a warning or not, as we would use `sct_run_batch` and perform prediction on 200 subjects (twice).

So, if I'm understanding correctly, we would expose this function in `sct_label_utils`, then people could include it in their `sct_run_batch` scripts as a processing step?

Reply:
I though abou it as more as an api, calls, but i guess it would make more sens

Reply:
Designing new API-specific features is tricky as I think [the API is not yet available](https://forum.spinalcordmri.org/t/using-sct-via-python/575/2?u=joshuacwnewton) to users... it's hard to anticipate what future usage could look like!

I was sort of thinking it could be good to leave this as a function and come back to this feature idea if it is high enough priority. (There are so many feature ideas in the issue backlog, and only so much time. :sweat_smile:)

Reply:
Yes I see what you mean. we can leave it here for now and we think it could be useful for the user we could add it pretty easily to sct_label_utils.